src/hex_tarball.erl

```diff
@@ -394,7 +395,12 @@ guess_build_tools(Metadata) ->
 %%====================================================================
 
 unpack_tarball(ContentsBinary, memory) ->
-    hex_erl_tar:extract({binary, ContentsBinary}, [memory, compressed]);
+    try hex_erl_tar:extract({binary, ContentsBinary}, [memory, compressed]) of
+      Ok -> Ok
+    catch
+      error:data_error ->
```

Comment:
Where does this error come from?

Reply:
```
iex> :zlib.gunzip("badgzip")
** (ErlangError) Erlang error: :data_error
    :zlib.inflate_nif(#Reference<0.540589784.484573189.118669>, 8192, 16384, 0)
    :zlib.dequeue_all_chunks_1/3
    :zlib.inflate/3
    :zlib.gunzip/1
```

one of the existing tests caught it. I thought about not handling it but I thought we should.

Reply:
just to be clear, to get this error someone would need to have a correct outer tarball but incorrectly gzipped inner tarball, so yeah, pretty uncommon.

Reply:
hmm, it's a change in behaviour:

on 23.1.1:

```
~% asdf shell erlang 23.1.1 ; erl -noshell -eval 'io:format("~p~n", [erl_tar:extract({binary, <<"bad">>}, [compressed])]), halt().'
{error,eof}
```

on master:

```
~% asdf shell erlang git ; erl -noshell -eval 'io:format("~p~n", [erl_tar:extract({binary, <<"bad">>}, [compressed])]), halt().'
{"init terminating in do_boot",{data_error,[{zlib,inflate_nif,[#Ref<0.1770301242.1831469062.81859>,8192,16384,0],[]},{zlib,dequeue_all_chunks_1,3,[]},{zlib,inflate,3,[]},{zlib,gunzip,1,[]},{erl_tar,open1,4,[{file,"erl_tar.erl"},{line,326}]},{erl_tar,foldl_read,4,[{file,"erl_tar.erl"},{line,1382}]},{erl_eval,do_apply,6,[{file,"erl_eval.erl"},{line,680}]},{erl_eval,expr,5,[{file,"erl_eval.erl"},{line,232}]}]}}
init terminating in do_boot ({data_error,[{zlib,inflate_nif,[,8192,16384,0],[]},{zlib,dequeue_all_chunks_1,3,[]},{zlib,inflate,3,[]},{zlib,gunzip,1,[]},{erl_tar,open1,4,[{_},{_}]},{erl_tar,foldl_read,

Crash dump is being written to: erl_crash.dump...done
```

so I think we should fix it in otp, I'll send a patch.

Reply:
Great find!

Reply:
Maybe add a note that the workaround is not needed after OTP 23.x?

Reply:
José already fixed it upstream :D https://github.com/erlang/otp/pull/2814 (I expect this to land as is, if there are any changes I'll re-apply them later.)